### ISSUE TITLE
Update IPDropDown comments to indicate portions of module loaded elsewhere

### ISSUE THREAD

**Issue by Brett208** [OutpostUniverse/op2ext#140]
Even when IPDropDown is disabled, the dropdown itself exists in game. I think another dll contains the framework of the UI where it is displayed. I'd like to document this behaviour in a comment inside IPDropDown becuase it first confused me and I thought we were not properly disabling it a while back when testing.

**Comment by DanRStevens** [comment]
Hmm, yes. I tried commenting out the lines that instantiate and register the module, yet the IP dropdown still appears in game. Very peculiar.

**Comment by Brett208** [comment]
I verified it doesn't actually work, just leaves the drop down (I've known about this for a while now), but thought we should document it. Maybe someday in the future we can set the drop down dynamically via memory edits, but I'm happy with just documenting for now.

**Comment by DanRStevens** [comment]
We should find out why this is happening. It doesn't make logical sense, so we should get to the bottom of it. Perhaps it's something weird, like the patch is being applied by another loaded module, like NetFix or NetHelper. Or maybe we weren't testing correctly. Or maybe part of the patch was baked into the exe with a hex editor.

**Comment by Brett208** [comment]
I suspect it is just a direct (baked in) hack in the shell.dll file to place a drop down on a couple of the screens. I am certain the IPDropDown patch is not being applied by NetFix or NetHelper or any other module based on my testing.

**Comment by Brett208** [comment]
I misspoke earlier in this issue. The dialogs are contained in out2res.dll and op2shres.dll. However, I did a cursory search of these dlls and did not see the dialog with the modified IP address field. So I'm not sure where they are being modified at?

**Comment by DanRStevens** [comment]
Add, thanks for the investigation. The dialog is in out2res.dll.

As a side note, I found a handy way of finding these things on Linux. There is a [`strings`](https://linux.die.net/man/1/strings) command, which can be used to scan binary files, such as EXE and DLL files, for printable strings. It has an encoding option so you can scan for wide strings, such as the UTF-16/UCS-2 strings used in Windows resources. Example:
```
strings --encoding l out2res.dll
```
The `l` stands for little-ending UTF-16.

Pipe that through `grep`, and you can quickly see if a given string from a user interface comes from a given DLL file:
```
strings --encoding l out2res.dll | grep "Looking at the IP address:"
```

Looks like there is a rough equivalent on Windows as well, with their own [`Strings`](https://docs.microsoft.com/en-us/sysinternals/downloads/strings) utility.

**Comment by Brett208** [comment]
Would you mind adding a bit of documentation on where it is in ipdropdown 's file(s).

I'm not 100% following what you found (which is fine).

**Comment by DanRStevens** [comment]
In the test above, I didn't have any information other than what file. I looked in game at the multiplayer join screen, and noted some text above the dropdown box for the IP. I then scanned the DLL files for that text, encoded as UTF-16. It came back with a match from out2rest.dll.

From what I remember, the dialog resources were edited to change the textbox to a combo box. This was done with a tool such as [Resource Hacker](http://www.angusj.com/resourcehacker/).

If you load up out2rest.dll with Resource Hacker, and check under "Dialog" for resource ID "2010", you'll see the "Find Game Session" dialog which contains the combo box for the IP address.

If you compare this to the resources found in an original distribution, such as the 1.2.0.7 release, you'll see that field was an Edit field instead.
